process/sync/metablock.go

```diff
@@ -386,13 +386,17 @@ func (boot *MetaBootstrap) doJobOnSyncBlockFail(hdr *block.MetaBlock, err error)
 		boot.requestsWithTimeout++
 	}
 
-	shouldRollBack := err != process.ErrTimeIsOut || boot.requestsWithTimeout >= process.MaxRequestsWithTimeoutAllowed
+	allowedRequestsWithTimeOutHaveReached := boot.requestsWithTimeout >= process.MaxRequestsWithTimeoutAllowed
+	isInProperRound := boot.rounder.Index()%roundModulusTrigger == 0
```

Comment:
I would have `boot.rounder.Index()%roundModulusTrigger` in a function since it is used in multiple places, it would also be easier to reuse  by others

Reply:
Yep, done